### DISCUSSION TITLE
오브젝트가 포물선 이동하도록 만들고싶습니다

### DISCUSSION THREAD

**Discussion by yoon-hw22** [naverz/zepeto-studio-kor#1051] [5. Q&A]
public targetPosition : GameObject;

    Update() {
        this.transform.position = Vector3.Slerp(this.gameObject.transform.position,this.targetPosition.transform.position,0.01);
    }
    
    이렇게 slerp함수를 사용했는데 그냥 직선으로만 이동하네요 
타입스크립트에서 slerp함수는 못사용하나요? 포물선으로 이동하게하려면 어떻게해야하나요

**Comment by YSHFM** [comment]
![image](https://user-images.githubusercontent.com/106574541/180339161-754e48f2-c7d1-4c23-8ea9-c09170642193.png)
https://terms.naver.com/entry.naver?docId=5741299&cid=60217&categoryId=60217
포물선 운동에 따라 포물선 이동 구현이 가능합니다.

**Reply by YSHFM** [reply]
g = 중력, v=파워 , t는 시간 정도로 이해하시면 되겠습니다.

**Reply by yoon-hw22** [reply]
직접 함수를 작성하는 방법 말고는 없나요?

**Reply by YSHFM** [reply]
slerp으로도 가능하겠습니다만 먼저 slerp은 구면선형보간으로 알고 있어서, 원하시는 포물선 운동을 구현하기에는 적합하지 않으실 수도 있습니다.

https://docs.unity3d.com/ScriptReference/Vector3.Slerp.html

slerp의 자세한 사용법은 유니티 공식문서에도 잘 설명이 되어 있어서, 해당 함수에 결과값에 따라 추가적인 함수제작을 한다면 포물선 모양이 나올 순 있으나, 포물선 운동식이 더 쉬울 수도 있습니다.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 가장 간단한 방법으로는
포물선으로 움직이고 싶은 오브젝트에 Rigidbody 컴포넌트를 추가한 후

```typescript
Start() {
        this.GetComponent<Rigidbody>().velocity = new Vector3(10.0, 8.0, 0.0);
}
```
와 같이 발사 당시의 힘을 정의해주면 포물선으로 날아갑니다.
